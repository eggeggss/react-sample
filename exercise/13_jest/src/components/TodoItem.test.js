import React from 'react';
import { shallow } from 'enzyme';

import TodoItem from './TodoItem';

const props = {
  id: 1,
  done: true,
  toggleItem: jest.fn(),
};

const text = 'text123';

const wrapper = shallow(<TodoItem {...props}>{text}</TodoItem>);

it('should render class', () => {

  expect(wrapper.hasClass('todo-item')).toBe(true);
  expect(wrapper.hasClass('done')).toBe(true);
});

// yarn run test

it('should render text', () => {
  expect(wrapper.text()).toBe(text);
});


it('should call toggleitem after clicked', () => {
  wrapper.simulate('click');
  expect(props.toggleItem).toHaveBeenCalledTimes(1);
});

it('should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});
