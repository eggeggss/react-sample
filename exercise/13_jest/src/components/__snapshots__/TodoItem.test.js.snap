// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<li
  className="todo-item done"
  onClick={[Function]}
>
  text123
</li>
`;
